### PR TITLE
[WIP] feat: webcomponent

### PR DESCRIPTION
**Pull request by nemi-notrace**
a repco recomender box

### CODE REVIEW

File: packages/repco-webcomponent/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (Frando) [inline packages/repco-webcomponent/.eslintrc.json:1]: can this file be removed? we already have a eslint setup. let's try to not add different setups to different packages.

File: packages/repco-webcomponent/.prettierrc.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (Frando) [inline packages/repco-webcomponent/.prettierrc.json:1]: as above: let's keep a single setup as with the other packages (config in root of repo)

File: packages/repco-webcomponent/package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "repco-recommander-box",
```
Comment (Frando) [inline packages/repco-webcomponent/package.json:2]: should be `repco-webcomponent` or rename the package folder, but i think `repco-webcomponent` is good

File: packages/repco-webcomponent/src/post-card/post-card-body.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import {css, html, LitElement} from 'lit';
+import {customElement, property} from 'lit/decorators.js';
+import {unsafeHTML} from 'lit/directives/unsafe-html.js';
+
+@customElement('post-card-body')
```
Comment (Frando) [inline packages/repco-webcomponent/src/post-card/post-card-body.ts:5]: we should prefix all custom elements with `repco-` to not have conflicts. custom elements share a global namespace per page unfortunately.

File: packages/repco-webcomponent/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (Frando) [inline packages/repco-webcomponent/tsconfig.json:1]: try to inherit from the top-level tsconfig, as in the other packages

File: package.json [hunk 6]
```diff
@@ -8,11 +8,13 @@
...
   "type": "module",
   "scripts": {
-    "build": "run-s build:prisma-generate build:prisma build:tsc build:cli-bundle",
+    "build": "run-s build:prisma-generate build:prisma build:tsc build:cli-bundle build:webcomponent",
     "build:tsc": "tsc --build",
     "build:prisma-generate": "yarn --cwd packages/repco-prisma-generate build",
     "build:prisma": "yarn --cwd packages/repco-prisma build",
     "build:cli-bundle": "yarn --cwd packages/repco-cli build:bundle",
+    "build:webcomponent": "yarn --cwd packages/repco-webcomponent build",
+    "webcomponent": "yarn --cwd packages/repco-webcomponent dev",
```
Comment (Frando) [inline package.json:17]: let's make this `dev:webcomponent`

File: packages/repco-webcomponent/LICENSE [hunk 7]
```diff
@@ -0,0 +1 @@
+tba
```
Comment (Frando) [inline packages/repco-webcomponent/LICENSE:1]: MIT and APACHE, as the whole repo.

File: packages/repco-webcomponent/README.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+```html
+<script type="module" src="./dist/src/post-element.js"></script>
+<repco-recommander-box></repco-recommander-box>
+```
+
+Use the `post-element` tag in your HTML file. You can customize the display of the component by setting its attributes:
+
+- `count`: the number of posts to display (default is `10` also maximum right now)
+- `defaultThumbnail`: the default thumbnail image URL to display if a post does not have a thumbnail (default is ` `)
```
Comment (Frando) [inline packages/repco-webcomponent/README.md:17]: this is `default-thumbnail` as HTML attribute (`defaultThumbnail`would be the DOM property then)

File: packages/repco-webcomponent/README.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+- `url`: the URL of the GraphQL API to retrieve the posts from (default is `https://node1.repco.openaudiosearch.org/graphql`)
```
Comment (Frando) [inline packages/repco-webcomponent/README.md:18]: let's rename this `endpoint` or `api-url`

File: packages/repco-webcomponent/src/post-card/post-card-body.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+      overflow: hidden;
+      text-overflow: ellipsis;
+      color: var(--medium-body-color);
+    }
+  `;
+
+  @property()
+  body = '';
+  override render() {
+    return html`${unsafeHTML(this.body)}`;
```
Comment (Frando) [inline packages/repco-webcomponent/src/post-card/post-card-body.ts:19]: this is .. unsafe ;) we likely need to check the HTML there. or only allow strings not html. we cannot inject any HTML coming from the repco node into the DOM, this opens it up to XSS etc vulnerabilities.

this is a bit of a bigger issue that we'll have to address in repco. for now, we should either sanitze the HTML here or convert it to string only (remove all html tags).

Comment (nemi-notrace) [reply packages/repco-webcomponent/src/post-card/post-card-body.ts:19]: See https://github.com/openaudiosearch/repco/issues/61

**Comment by nemi-notrace** [comment]
moved to oas

**Pull request closed without merging**